packages/sit-onyx/src/composables/useWedgePosition.ts

```diff
@@ -0,0 +1,80 @@
+import { ref, unref, type MaybeRef } from "vue";
+
+export type WedgePosition = "center" | "left" | "right";
+
+export const useWedgePosition = (
+  element: MaybeRef<Element | undefined>,
+  tooltipElement: MaybeRef<Element | undefined>,
+) => {
+  const minMargin = 16;
+  const wedgePosition = ref<WedgePosition>("center");
+
+  const updateWedgePosition = () => {
+    const wrapperEl = unref(element);
+    const tooltipEl = unref(tooltipElement);
+
+    if (!wrapperEl || !tooltipEl) {
+      wedgePosition.value = "center";
+      return;
+    }
+
+    const overflowParentRect = findParentWithHiddenOverflow(wrapperEl)?.getBoundingClientRect();
+    const wrapperRect = wrapperEl.getBoundingClientRect();
+    const tooltipElementRect = tooltipEl.getBoundingClientRect();
+
+    if (tooltipElementRect.width < wrapperRect.width) {
+      wedgePosition.value = "center";
+      return;
+    }
+
+    const minSpace = (tooltipElementRect.width - wrapperRect.width + minMargin * 2) / 2;
+
+    const parentLeft = overflowParentRect?.left ?? window.visualViewport?.pageLeft ?? 0;
+    const parentRight =
+      overflowParentRect?.right ??
+      (window.visualViewport?.width ?? window.innerWidth) +
+        (window.visualViewport?.offsetLeft ?? 0) ??
+      0;
+
+    const freeSpaceLeft = wrapperRect.left - parentLeft;
+    const freeSpaceRight = parentRight - wrapperRect.right;
+
+    const enoughSpaceLeft = freeSpaceLeft >= minSpace;
+    const enoughSpaceRight = freeSpaceRight >= minSpace;
+
+    wedgePosition.value =
+      enoughSpaceLeft === enoughSpaceRight
+        ? "center"
+        : freeSpaceLeft > freeSpaceRight
+          ? "right"
+          : "left";
+  };
+
+  /**
+   * Recursively finds the first parent element with hidden overflow.
+   */
+  const findParentWithHiddenOverflow = (element?: Element): Element | undefined => {
```

Comment:
This is copied from the `useOpenDirection` composable, right? So we could export it there to not duplicate code